sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -274,6 +274,8 @@ private[hive] class HiveClientImpl(
     val original = Thread.currentThread().getContextClassLoader
     // Set the thread local metastore client to the client associated with this HiveClientImpl.
     Hive.set(client)
+    // replace conf in cached hive with current conf
```

Comment:
A `get` after a `set` looks weird. Actually I need to go back to check Hive source to know `get` will replace conf with provided one. Shall we move this into `private def client: Hive` and add comment for it?